src/TSTransformer/util/addDiagnosticIfNotCached.ts

```diff
@@ -0,0 +1,23 @@
+import { TransformState } from "TSTransformer";
+import { DiagnosticService } from "TSTransformer/classes/DiagnosticService";
+import ts from "typescript";
+
+export function addDiagnosticIfNotCached(symbol: ts.Symbol, error: ts.Diagnostic, cache: Set<ts.Symbol>) {
+	if (!cache.has(symbol)) {
+		cache.add(symbol);
+		DiagnosticService.addDiagnostic(error);
+	}
+}
+
+export function addDiagnosticFromNodeIfNotCached(
```

Comment:
This might fit better as a method of DiagnosticService

Reply:
Good point, will move.